### PR TITLE
[W5.7][T12-4]Changrui Li

### PR DESCRIPTION
**Pull request by leo-1997**

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -72,6 +72,11 @@ Examples:
 Shows a list of all persons, along with their non-private details, in the address book. +
 Format: `list`
 
+== Sorting all persons : `sort`
+
+Shows a list of all persons, according to their in the alphabetical order, along with their non-private details, in the address book. +
+Format: `list`
+
```
Comment (YShu7) [inline docs/UserGuide.adoc]: Typo here. Should be `sort`.

File: test/expected.txt [hunk 2]
```diff
@@ -297,6 +299,33 @@
...
+|| Enter command: || [Command entered:  add Betsy Choo pp/222222 pe/benchoo@nus.edu.sg pa/222, beta street t/secretive]
+|| New person added: Betsy Choo Phone: (private) 222222 Email: (private) benchoo@nus.edu.sg Address: (private) 222, beta street Tags: [secretive]
+|| ===================================================
+|| Enter command: || [Command entered:  sort]
+|| 	1. Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street Tags: 
+|| 	2. Betsy Choo Tags: [secretive]
+|| 	3. Charlie Dickson Email: charlie.d@nus.edu.sg Address: 333, gamma street Tags: [school][friends]
+|| 	4. Dickson Ee Phone: 444444 Address: 444, delta street Tags: [friends]
+|| 	5. Esther Potato Phone: 555555 Email: esther@not.a.real.potato Tags: [tubers][starchy]
+||
```
Comment (YShu7) [inline test/expected.txt]: Because I notice you're using `CASE_INSENSITIVE_ORDER` in sort command, I think you may wan to add someone whose name starts with lower case letters to make the test cases more various.

File: docs/UserGuide.adoc (hunk 1 above)
Comment (leo-1997) [reply docs/UserGuide.adoc]: Thank you! I have corrected the typo.

File: test/expected.txt (hunk 2 above)
Comment (leo-1997) [reply test/expected.txt]: Thank you! I have updated to test the CASE_INSENSITIVE_ORDER in sort command.

File: src/seedu/addressbook/commands/HelpCommand.java [hunk 3]
```diff
@@ -15,14 +15,15 @@
...
-                + "\n" + ExitCommand.MESSAGE_USAGE
+                        + "\n" + DeleteCommand.MESSAGE_USAGE
+                        + "\n" + ClearCommand.MESSAGE_USAGE
+                        + "\n" + FindCommand.MESSAGE_USAGE
+                        + "\n" + ListCommand.MESSAGE_USAGE
+                        + "\n" + SortCommand.MESSAGE_USAGE
+                        + "\n" + ViewCommand.MESSAGE_USAGE
+                        + "\n" + ViewAllCommand.MESSAGE_USAGE
+                        + "\n" + HelpCommand.MESSAGE_USAGE
+                        + "\n" + ExitCommand.MESSAGE_USAGE
```
Comment (eugenepeh) [inline src/seedu/addressbook/commands/HelpCommand.java:26]: I believe the original indentation is fine? 
Be careful of unnecessary or accidental changes here as it can cause unnecessary merge conflict.

File: src/seedu/addressbook/parser/Parser.java [hunk 4]
```diff
@@ -34,7 +35,8 @@
     public static final Pattern KEYWORDS_ARGS_FORMAT =
             Pattern.compile("(?<keywords>\\S+(?:\\s+\\S+)*)"); // one or more keywords separated by whitespace
 
-    public static final Pattern PERSON_DATA_ARGS_FORMAT = // '/' forward slashes are reserved for delimiter prefixes
+    public static final Pattern PERSON_DATA_ARGS_FORMAT = // '/' forward slashes are reserved for delimiter
+            // prefixes
```
Comment (eugenepeh) [inline src/seedu/addressbook/parser/Parser.java:39]: similarly, is this change necessary?

File: src/seedu/addressbook/commands/SortCommand.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        ArrayList<ReadOnlyPerson> sortAllPersons = new ArrayList<>((allPersons));
+        sortAllPersonsInAlphabeticalOrder(sortAllPersons);
+        return new CommandResult(getMessageForPersonListShownSummary(sortAllPersons), sortAllPersons);
+    }
+
+    /**
+     * Sort the list of persons according to their names in alphabetical order.
+     */
+    private static void sortAllPersonsInAlphabeticalOrder(List<ReadOnlyPerson> allPersons) {
+        Collections.sort(allPersons, new Comparator<>() {
```
Comment (eugenepeh) [inline src/seedu/addressbook/commands/SortCommand.java:31]: good, another way would be to use lambda expression.

**Review by eugenepeh — APPROVED** [review]
Good job on completing the activity.

You've updated all the aspects, docs, tests etc...
However take note that there are quite a few accidental changes that shouldn't be there, also the indentation. 

Some comments added. Please close the PR after reading the comments.
